src/test/java/jenkins/plugins/logstash/persistence/ElasticSearchDaoTest.java

```diff
@@ -0,0 +1,127 @@
+package jenkins.plugins.logstash.persistence;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.HttpClientBuilder;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.io.PrintStream;
+import java.net.SocketException;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.*;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ElasticSearchDaoTest {
+  ElasticSearchDao dao;
+  @Mock HttpClientBuilder mockClientBuilder;
+  @Mock CloseableHttpClient mockHttpClient;
+  @Mock HttpPost mockPost;
+  @Mock CloseableHttpResponse mockResponse;
+  @Mock PrintStream mockLogger;
+
+  @Before
+  public void before() throws Exception {
+    int port = (int) (Math.random() * 1000);
+    dao = new ElasticSearchDao("localhost", port, "logstash", "username", "password");
+
+    // Note that we can't run these tests in parallel
+    dao.clientBuilder = mockClientBuilder;
+
+    when(mockClientBuilder.build()).thenReturn(mockHttpClient);
+
+    when(mockHttpClient.execute(mockPost)).thenReturn(mockResponse);
+  }
+
+  @After
+  public void after() throws Exception {
+    verifyNoMoreInteractions(mockClientBuilder);
+    verifyNoMoreInteractions(mockHttpClient);
+    verifyNoMoreInteractions(mockPost);
+    verifyNoMoreInteractions(mockLogger);
+  }
```

Comment:
You have some unused imports that can be cleaned up:

import org.apache.http.entity.StringEntity;
import org.mockito.Matchers;
import java.net.SocketException;